src/api/v4/layer/layer.js

```diff
@@ -1,90 +1,274 @@
 var Base = require('./base');
 var CartoDBLayer = require('../../../geo/map/cartodb-layer');
+var SourceBase = require('../source/base');
+var StyleBase = require('../style/base');
 
 /**
- *
  * Represent a layer Object.
+ * 
+ * @param {object} source - The source where the layer will fetch the data.
+ * @param {carto.style.CartoCSS} style - A CartoCSS object with the layer styling.
+ * @param {object} [options]
+ * @param {Array<string>} [options.featureClickColumns=[]] - Columns that will be available for `featureClick` events. 
+ * @param {Array<string>} [options.featureOverColumns=[]] - Columns that will be available for `featureOver` events.
  *
- * @param {string} [id] - A unique ID for this layer
- * @param {carto.source.Base} source - The source where the layer will fetch the data.
- * @param {carto.style.Base} style - A CartoCSS object with the layer styling.
- *
+ * @fires carto.layer.Layer.FeatureEvent
  * @example
  *
- * new carto.layer.Layer('cities', citiesSource, citiesStyle);
+ * // no options
+ * new carto.layer.Layer(citiesSource, citiesStyle);
  *
  * @example
  *
- * new carto.layer.Layer(citiesSource, citiesStyle);
- *
+ * // with options
+ * new carto.layer.Layer(citiesSource, citiesStyle, {
+ *   featureClickColumns: [ 'name', 'population' ],
+ *   featureOverColumns: [ 'name' ]
+ * });
+ * 
  * @constructor
- * @extends carto.layer.Base
  * @memberof carto.layer
  * @api
  */
-function Layer (id, source, style) {
-  if (typeof style === 'undefined') {
-    source = id;
-    style = source;
-    id = 'fakeId'; // TODO: Generate a unique ID
-  }
+function Layer (source, style, options) {
+  options = options || {};
+
+  _checkSource(source);
+  _checkStyle(style);
+
+  this._engine = undefined;
+  this._internalModel = undefined;
 
-  this._id = id;
   this._source = source;
   this._style = style;
   this._visible = true;
+  this._featureClickColumns = options.featureClickColumns || [];
```

Comment:
`options` can be empty and throw an error in this case.

Reply:
We're making sure options is not "empty" with `options = options || {};`. When this happens `options.featureClickColumns` will be `undefined` and `this._featureClickColumns` will be initialised as a `[]`. Are we missing some other case?